src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -2586,42 +2596,651 @@ class C
+            };
+
+            var comp = CompileAndVerify(source, expectedOutput: expected, parseOptions: TestOptions.Regular, sourceSymbolValidator: validator);
+            comp.VerifyDiagnostics();
+        }
+
+        private static void VerifyModelForDeconstructionLocal(SemanticModel model, VariableDeclarationSyntax decl, params IdentifierNameSyntax[] references)
+        {
+            var variableDeclaratorSyntax = decl.Variables.Single();
+            var symbol = model.GetDeclaredSymbol(variableDeclaratorSyntax);
+            Assert.Equal(variableDeclaratorSyntax.Identifier.ValueText, symbol.Name);
+            Assert.Equal(LocalDeclarationKind.RegularVariable, ((LocalSymbol)symbol).DeclarationKind);
+            Assert.Same(symbol, model.GetDeclaredSymbol((SyntaxNode)variableDeclaratorSyntax));
+            Assert.Same(symbol, model.LookupSymbols(decl.SpanStart, name: variableDeclaratorSyntax.Identifier.ValueText).Single());
+            Assert.True(model.LookupNames(decl.SpanStart).Contains(variableDeclaratorSyntax.Identifier.ValueText));
+
+            var local = (SourceLocalSymbol)symbol;
+
+            if (local.IsVar && local.Type.IsErrorType())
+            {
+                Assert.Null(model.GetSymbolInfo(decl.Type).Symbol);
+            }
+            else
+            {
+                if (decl.Type != null)
+                {
+                    Assert.Equal(local.Type, model.GetSymbolInfo(decl.Type).Symbol);
+                }
+            }
+
+            foreach (var reference in references)
+            {
+                Assert.Same(symbol, model.GetSymbolInfo(reference).Symbol);
+                Assert.Same(symbol, model.LookupSymbols(reference.SpanStart, name: variableDeclaratorSyntax.Identifier.ValueText).Single());
+                Assert.True(model.LookupNames(reference.SpanStart).Contains(variableDeclaratorSyntax.Identifier.ValueText));
+                Assert.Equal(local.Type, model.GetTypeInfo(reference).Type);
+            }
+        }
+
+        private static VariableDeclarationSyntax GetDeconstructionLocal(SyntaxTree tree, string name)
+        {
+            return tree.GetRoot().DescendantNodes().OfType<VariableDeclarationSyntax>().Where(p => p.Variables.Count() == 1 && p.Variables.Single().Identifier.ValueText == name).Single();
+        }
+
+        private static IdentifierNameSyntax GetReference(SyntaxTree tree, string name)
+        {
+            return GetReferences(tree, name).Single();
+        }
+
+        private static IEnumerable<IdentifierNameSyntax> GetReferences(SyntaxTree tree, string name)
+        {
+            return tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == name);
+        }
+
+        [Fact]
+        public void DeclarationWithActualVarType()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        (var x1, int x2) = (new var(), 2);
+        var (x3, x4) = (3, new var());
```

Comment:
See previous comment for more details, but my thinking was that no type is allowed in that position, so "var" is un-ambiguous and always means "type to be inferred".

Even if we wanted to allow types in that deconstruction form in the future, the most likely syntax that we discussed was `(int, string) (x1, x2) = ...`, not `int (x1, x2) = ...`, if I remember the discussion correctly. 

Thanks for the suggestion of alias tests in semantic model. I'll add. Another var ambiguity test on my list is having a ref-returning var method.